paperspace/cli.py

```diff
@@ -0,0 +1,208 @@
+import functools
+import json
+
+import click
+
+from paperspace import commands, constants
+
+
+class ChoiceType(click.Choice):
+    """Takes a string-keyed map and converts cli-provided parameter to corresponding value"""
+
+    def __init__(self, type_map, case_sensitive=True):
+        super(ChoiceType, self).__init__(tuple(type_map.keys()), case_sensitive=case_sensitive)
+        self.type_map = type_map
+
+    def convert(self, value, param, ctx):
+        value = super(ChoiceType, self).convert(value, param, ctx)
+        return self.type_map[value]
+
+
+EXPERIMENT_TYPES_MAP = {
```

Comment:
EXPERIMENT_TYPE_SINGLE_NODE_ID = 1
EXPERIMENT_TYPE_SINGLE_NODE = "single"
EXPERIMENT_TYPE_GRPC_MULTI_NODE_ID = 2
EXPERIMENT_TYPE_GRPC_MULTI_NODE = "multi-grpc"
EXPERIMENT_TYPE_MPI_MULTI_NODE_ID = 3
EXPERIMENT_TYPE_MPI_MULTI_NODE = "multi-mpi"

EXPERIMENT_TYPES = {
    EXPERIMENT_TYPE_SINGLE_NODE: EXPERIMENT_TYPE_SINGLE_NODE_ID,
    EXPERIMENT_TYPE_GRPC_MULTI_NODE: EXPERIMENT_TYPE_GRPC_MULTI_NODE_ID,
    EXPERIMENT_TYPE_MPI_MULTI_NODE: EXPERIMENT_TYPE_MPI_MULTI_NODE_ID,
}

Reply:
I don't get it. There is no need for single - this constant is used only for multinode (but It seems I should change it's name)

As for the rest - "multi" in "multi-grpc" is not needed since it's used only for multi(node)

Reply:
Yeah, for now ;) I also think it won't hurt to have it set, for instance for those to come to know

Reply:
OMG, guys please read the context. This constant is used in one place only and it is to choose between two multinode modes that are currently supported.